### DISCUSSION TITLE
Curation & Moderation in the Architecture

### DISCUSSION THREAD

**Discussion by bobwyman** [bluesky-social/atproto#105] [General Discussion]
I am concerned that the architecture overview shows "Curation and Moderation" (C&M) as a single function that is located between the Speech and Reach layers. This particular graphic representation may lead some to believe that C&M is a monolithic function that exists primarily or exclusively in the channel. Thus, the erroneous idea that C&M must always be performed by a third-party may be implied. It would be more flexible and correct to suggest that C&M is a function that can be performed by any of a speaker, a channel component, or a recipient.

I am also concerned with this statement:
> "Moderation occurs during aggregation or by end-user choice of who they follow and block."

As written, it seems to significantly limit where, when, and how C&M may occur. Moderation may occur outside the context of "aggregation" (e.g. in the context of something like a Direct Message), and end-users shouldn't be limited to only "follow and block" methods for curating what they see. Certainly, the end user should have significant choice over the method and means for C&M, however, limiting that choice is both unnecessary and just bad.

bob wyman

**Comment by dholms** [comment]
Hey Bob, thanks for the post.

You're correct that any actor in the system can moderate & curate content within the scope that they act. So, for instance, a user may filter some content client-side before displaying it, or a PDS could opt not to host content from a given user. In fact data servers will necessarily be moderating content to some degree insofar as they are not hosting illegal content in their jurisdiction. 

Since this is an open & distributed system, users and services have the freedom to choose the software that they'll run - as long as it adheres to the specified APIs. We aren't limiting a PDS or user in terms of their moderation capabilities. However we expect a lot of  moderation & curation to occur at the aggregation level, which is why the focus is drawn to that layer.

**Reply by dholms** [reply]
We updated how we express this in the architecture overview: https://github.com/bluesky-social/adx/pull/107/files

😁

**Reply by bobwyman** [reply]
Thanks for that rapidly implemented clarification!
I really do think it is important to ensure that people understand that "user choice" means that they should be able to do their own Curation & Moderation and that it should be possible to do, but potentially difficult or expensive, even if no other shared system component is assisting the process. While virtually everyone will undoubtedly choose to rely at least partially on some shared C&M component, we should understand that the system can, and should, work properly even if only private C&M components and systems are in use.

This should also include the idea that there may be multiple C&M components in the "path" between a speaker and a recipient. I may, for instance, rely on a shared C&M provider to do most of the work for me, but then use private C&M functions (such as a locally implemented block list) to impose a second, private layer of C&M. I might also ask two C&M providers to label content and then have a third, private component that takes actions based on the combined set of labels. (e.g. The two labelers might apply different kinds of labels. One might apply labels indicating authors' membership in particular groups, such members of a press association, while the other might label content based on its subject. Given both sets of labels, I could then filter for "members of the press" writing about "BlueSky" but nothing written by "Bob OR Alice"...)

bob wyman

**Reply by dholms** [reply]
That's exactly how we're thinking about it as well. The system should be very flexible with options for moderation at every step - with the final option for local block lists on the client. And yup you can combine labels from multiple labelers. That's one of the benefits to splitting labeling out from moderation. You can mix & match or weight labels from different services differently. This architecture write up is a whirlwind introduction to every level of the system, but we'll be doing deep dives into certain parts (C&M will likely be the first!) here in the near future

**Comment by bobwyman** [comment]
Also, I'd like to point out that the [W3C Credible Web Community Group](https://credweb.org/) has been grappling with the problem of standardizing ways to label the credibility of authors, items, and parts of items. Although the effort has been less vigorous than I would like, my hope is that we'll be able to develop a nexus between your ADX work and that of the CredWeb group.

bob wyman
